src/modules/hip/kernel/normalize.hpp

```diff
@@ -0,0 +1,1925 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+#define MAX_SHARED_MEMORY_SIZE 1024
+
+__device__ __forceinline__ uint rpp_hip_mod(uint a, uint b)
+{
+    return (a % b);
+}
+
+__device__ uint compute_2d_paramindex(uint y, uint x, uint *paramShape, uint *paramStrides)
+{
+    uint yFactor =  ((paramShape[0] > 1)) ? (rpp_hip_mod(y, paramShape[0])) * paramStrides[0] : 0;
+    uint xFactor =  ((paramShape[1] > 1)) ? (rpp_hip_mod(x, paramShape[1])) * paramStrides[1] : 0;
+    uint paramIndex = yFactor + xFactor;
+    return paramIndex;
+}
+
+__global__ void normalize_2d_hip_tensor(float *srcPtr,
+                                        uint2 srcStridesNH,
+                                        float *dstPtr,
+                                        uint2 dstStridesNH,
+                                        float *meanTensor,
+                                        float *stdDevTensor,
+                                        float scale,
+                                        float shift,
+                                        uint *roiTensor,
+                                        uint maxParamVolume,
+                                        uint axisMask)
+{
+    uint id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x; // width
+    uint id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y; // height
+    uint id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z; // batchsize
+
+    uint *roi = &roiTensor[id_z * 4 + 2];
+    uint height = roi[0];
+    uint width = roi[1];
+
+    if (id_x >= width || id_y >= height)
+        return;
+
+    uint paramIndex = id_z * maxParamVolume;
+    // update paramIndex based on axisMask value
+    if(axisMask == 1)
+        paramIndex += id_x;
+    else if(axisMask == 2)
+        paramIndex += id_y;
+
+    uint srcIdx = (id_z * srcStridesNH.x) + (id_y * srcStridesNH.y) + id_x;
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x;
+    float mean = meanTensor[paramIndex];
+    float stdDev = stdDevTensor[paramIndex];
+    float stdDevSquare = stdDev * stdDev;
+    float invStdDev = stdDevSquare ? rsqrtf(stdDevSquare) * scale : 0;
+    dstPtr[dstIdx] = fmaf((srcPtr[srcIdx] - mean), invStdDev, shift);
+}
+
+// __device__ void normalize_hip_compute(d_float8 *data_f8, d_float8 *mean_f8, d_float8 *invStdDev_f8, d_float8 *shift_f8)
```

Comment:
Remove L58-L137 if not needed

Reply:
I have kept it for the reference since 8 pixel vectorized is performing bad compared to single pixel versio